src/Raven.Studio/typescript/viewmodels/database/query/spatialQueryMap.ts

```diff
@@ -0,0 +1,142 @@
+                                     .bindPopup(generatePopupContent(point.PopupContent), { 'className' : 'custom-popup', 'maxWidth': 600, 'maxHeight': 400 } );
+                markers.push(pointMarker);
+            });
+
+            markersGroup.addLayers(markers);
+            const numberOfMarkersInLayer = markers.length;
+            
+            const controlLayerText = `<span>Point fields: <span class="number-of-markers pull-right padding padding-xs">${numberOfMarkersInLayer}</span></span>
+                                      <div class='margin-left'>
+                                          <span>${genUtils.escapeHtml(markersLayer.latitudeProperty)}</span><br>
+                                          <span>${genUtils.escapeHtml(markersLayer.longitudeProperty)}</span>
+                                      </div>`;
+
+            // A markersGroup layer is per spatial point from query
+            (dataLayers as any)[controlLayerText] = markersGroup;
+            markersGroups.push(markersGroup);
+        })
+        
+        const polyArray: Layer[] = [];
+        for (let i = 0; i < this.polygonsLayer().length; i++) {
+            const poly = this.polygonsLayer()[i];
+            const polyItem = L.polygon(poly.vertices,
+                { color: spatialPolygonModel.colors[i % spatialPolygonModel.colors.length] });
+            polyArray.push(polyItem);
+        }
+
+        const circleArray: Layer[] = [];
+        for (let i = 0; i < this.circlesLayer().length; i++) {
+            const circle = this.circlesLayer()[i];
+            const circleItem = L.circle([circle.latitude, circle.longitude],
+                { color: spatialCircleModel.colors[i % spatialCircleModel.colors.length], fillOpacity: 0.4, radius: circle.radius });
+            circleArray.push(circleItem);
+        }
+        
+        const polyLayer = L.layerGroup(polyArray);
+        const circleLayer = L.layerGroup(circleArray);
+        
+        if (polyArray.length) {
+            (dataLayers as any)["Polygons"] = polyLayer;
+        }
+        
+        if (circleArray.length) {
+            (dataLayers as any)["Circles"] = circleLayer;
+        }
+
+        // Must init L.map w/ some options, otherwise method Circle.getBounds() will fail
+        // The map.fitBounds method that is called later overrides these options
+        const map = L.map("mapid", {center:[0, 0], zoom: 1, preferCanvas: true});
+
+        osmMap.addTo(map);
+        L.control.layers(baseLayers, dataLayers).addTo(map);
+        
+        markersGroups.forEach(group => map.addLayer(group));
+        map.addLayer(polyLayer);
+        map.addLayer(circleLayer);
+
+        const mapBounds = L.latLngBounds([]);
+        markersGroups.forEach(group => mapBounds.extend(group.getBounds()));
+       
+        polyArray.forEach(poly => mapBounds.extend((poly as L.Polygon).getBounds()));
```

Comment:
do we need cast here? Maybe both methods has getBounds method?

Reply:
done